plugin/action/throttle/redis_limiter.go

```diff
@@ -180,14 +194,62 @@ func (l *redisLimiter) updateLimiterValues(maxID, bucketIdx int, totalLimiterVal
 	l.totalLimiter.mu.Unlock()
 }
 
+func getLimitValFromJson(data []byte, valField string) (int64, error) {
+	var limitVal any
+	var v int64
+	var err error
+	var m map[string]any
+	if err = json.Unmarshal(data, &m); err != nil {
+		return v, fmt.Errorf("failed to unmarshal map: %w", err)
+	}
+	limitVal, has := m[valField]
+	if !has {
+		return v, fmt.Errorf("no %q key in map", valField)
+	}
+	switch val := limitVal.(type) {
+	case string:
+		if v, err = strconv.ParseInt(val, 10, 64); err != nil {
+			return v, fmt.Errorf("failed to convert value to int64: %v", err)
+		}
+	case float64:
+		v = int64(val)
+	default:
+		return v, fmt.Errorf("invalid type of val in map under %q key", valField)
+	}
+	return v, nil
+}
+
 // updateKeyLimit reads key limit from redis and updates current limit.
-func (l *redisLimiter) updateKeyLimit() {
+func (l *redisLimiter) updateKeyLimit() error {
 	// try to set global limit to default
-	if b, err := l.redis.SetNX(l.keyLimit, l.totalLimiter.limit.value, 0).Result(); err == nil && !b {
+	val := strconv.FormatInt(l.totalLimiter.limit.value, 10)
+	if l.valField != "" {
+		val = fmt.Sprintf(`{"%s":"%s"}`, l.valField, val)
```

Comment:
Do we need escape `limiter_value_field`?
```suggestion
	key, _ := json.Marshal(l.valField) // todo handle error, do it once in the `.Start`
	val = fmt.Sprintf(`{"%s":"%s"}`, key, val)
```

Reply:
done, moved to new limiter creation func